java-runtime-builder-app/src/test/java/com/google/cloud/runtimes/builder/config/domain/BuildContextTest.java

```diff
@@ -87,14 +89,36 @@ public void testWriteDockerFilesWithExistingDockerignoreTerminalNewline() throws
     assertEquals(expectedDockerIgnore, readFile(getDockerIgnore()));
   }
 
-  @Test(expected = IllegalStateException.class)
+  @Test
   public void testWriteDockerFilesWithExistingDockerfile() throws IOException {
-    workspace = new TestWorkspaceBuilder()
-        .file("Dockerfile").withContents("FROM foo\n").build()
-        .build();
 
-    BuildContext context = initBuildContext();
-    context.writeDockerResources();
+    List<String[]> dockerPathsMessages = ImmutableList.of(
+        new String[]{"Dockerfile",
+            "Custom Dockerfiles aren't supported. If you wish to use a custom Dockerfile, consider "
+                + "using runtime: custom. Otherwise, remove the Dockerfile from the root "
+                + "to continue."},
+        new String[]{"src/main/docker/Dockerfile",
+            "Custom Dockerfiles aren't supported. If you wish to use a custom Dockerfile, consider "
+                + "using runtime: custom. Otherwise, remove the Dockerfile at "
+                + "src/main/docker/Dockerfile to continue."});
+
+    int errorMessagesMatched = 0;
```

Comment:
I would prefer two separate test methods, although you can create a utility method to reuse code.
Otherwise, it's not clear which of the two tests fails because all you'll see is that number of expected errors not matched.